### PR TITLE
feat: add functions/esm sample

### PR DESCRIPTION
**Pull request by grant**
## Functions ESM Sample

Adds a sample showing how to use ES modules within Google Cloud Functions for the Node.js 12+ runtime.

### Features

- `README.md`
  - Sample title, summary, prereqs
  - Running instructions
  - Testing instructions
- `const.js` file showing how to use `export` keyword.
- `index.js` file showing how to import a file and export a user function.
- `package.json` showing how to use `"type": "module"`
- Test

This sample would be a section of the Node.js Runtime doc showing how to use the popular Node feature of ES modules.

### CODE REVIEW

**Review by bcoe — APPROVED** [review]
This is slick 😄

**Review by ace-n — APPROVED** [review]
LGTM once tests pass.

I assume this will go on the [Node.js Runtime page](https://cloud.google.com/functions/docs/concepts/nodejs-runtime)?

**Review by ace-n — CHANGES_REQUESTED** [review]
FYI - @iennae is looking for a firm owner for this.

While I think the sample itself is worth adding, we need to confirm (as a team) that we have a plan to support it.

File: functions/esm/index.js
Comment (grayside) [inline functions/esm/index.js:17]: @bcoe I recall you saying the Node.js libraries weren't ready to support ESM yet, is that still the case? It seems okay if functions gets out ahead of the rest of Cloud, but we'll want to make the caveats highly visible if so.

In that case, I think this would need comments with that warning, in addition to carefully highlighting in the docs.

I assume if this is working the Functions Framework is ESM-ready?

**Comment by grant** [comment]
@JustinBeckwith Hi Justin, I see you added the `do not merge` label, but didn't comment as to why, which is pretty surprising. Why was it added? Can I remove the label?

**Comment by JustinBeckwith** [comment]
I entirely defer this one to @ace-n

**Comment by ace-n** [comment]
Hi folks,

@iennae and myself are working on figuring out an ownership/maintenance plan for this sample.

Let's hold off on merging this until we (the Serverless DPE team) have such a plan approved and in place.

**Comment by iennae** [comment]
Hey, thanks everyone. Just adding the flag back so we are clear to not merge currently. 

We appreciate the sample Grant. Serverless DPE will do the same due diligence as with all external sample requests. We'll get back to you after we review with the team on next steps.

**Comment by grant** [comment]
Alright, thanks. Will wait. LMK if I can do anything to help support this and our developers.

**Comment by grant** [comment]
Per sync, we'll lead developers who want to learn more about ESM with Cloud Functions to the Functions Framework and docs.

**Pull request closed without merging**
